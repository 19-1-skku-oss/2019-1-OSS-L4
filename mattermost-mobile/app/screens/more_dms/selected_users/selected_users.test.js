// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {shallow} from 'enzyme';

import Preferences from 'mattermost-redux/constants/preferences';

import SelectedUsers from './selected_users.js';

describe('SelectedUsers', () => {
    const baseProps = {
        onRemove: jest.fn(),
        maxCount: 3,
        profiles: {
            userId1: {
                id: 'userId1',
            },
            userId2: {
                id: 'userId2',
            },
            userId3: {
                id: 'userId3',
            },
        },
        selectedIds: {
            userId1: true,
        },
        theme: Preferences.THEMES.default,
        teammateNameDisplay: 'full_name',
        warnCount: 1,
    };

    test('should match snapshot', () => {
        const wrapper = shallow(
            <SelectedUsers {...baseProps}/>
        );

        expect(wrapper.getElement()).toMatchSnapshot();
    });

    test('should match snapshot to show warning for ability to add one more user', () => {
        const props = {
            ...baseProps,
            selectedIds: {
                userId2: true,
                userId1: true,
            },
        };

        const wrapper = shallow(
            <SelectedUsers {...props}/>
        );

        expect(wrapper.getElement()).toMatchSnapshot();
    });

    test('should match snapshot for no warning message', () => {
        const props = {
            ...baseProps,
            warnCount: 2,
        };

        const wrapper = shallow(
            <SelectedUsers {...props}/>
        );

        expect(wrapper.getElement()).toMatchSnapshot();
    });
});
