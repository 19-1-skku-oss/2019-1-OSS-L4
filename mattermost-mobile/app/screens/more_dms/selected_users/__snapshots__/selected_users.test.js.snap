// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectedUsers should match snapshot 1`] = `
<View
  style={
    Object {
      "marginBottom": 5,
      "marginLeft": 5,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "flexDirection": "row",
        "flexWrap": "wrap",
      }
    }
  >
    <SelectedUser
      onRemove={[MockFunction]}
      teammateNameDisplay="full_name"
      theme={
        Object {
          "awayIndicator": "#ffbc42",
          "buttonBg": "#166de0",
          "buttonColor": "#ffffff",
          "centerChannelBg": "#ffffff",
          "centerChannelColor": "#3d3c40",
          "codeTheme": "github",
          "dndIndicator": "#f74343",
          "errorTextColor": "#fd5960",
          "linkColor": "#2389d7",
          "mentionBj": "#ffffff",
          "mentionColor": "#145dbf",
          "mentionHighlightBg": "#ffe577",
          "mentionHighlightLink": "#166de0",
          "newMessageSeparator": "#ff8800",
          "onlineIndicator": "#06d6a0",
          "sidebarBg": "#145dbf",
          "sidebarHeaderBg": "#1153ab",
          "sidebarHeaderTextColor": "#ffffff",
          "sidebarText": "#ffffff",
          "sidebarTextActiveBorder": "#579eff",
          "sidebarTextActiveColor": "#ffffff",
          "sidebarTextHoverBg": "#4578bf",
          "sidebarUnreadText": "#ffffff",
          "type": "Mattermost",
        }
      }
      user={
        Object {
          "id": "userId1",
        }
      }
    />
  </View>
  <FormattedText
    defaultMessage="You can add {remaining, number} more users"
    id="mobile.more_dms.add_more"
    style={
      Object {
        "color": "rgba(61,60,64,0.6)",
        "fontSize": 12,
        "marginBottom": 2,
        "marginRight": 5,
        "marginTop": 10,
      }
    }
    values={
      Object {
        "remaining": 2,
      }
    }
  />
</View>
`;

exports[`SelectedUsers should match snapshot for no warning message 1`] = `
<View
  style={
    Object {
      "marginBottom": 5,
      "marginLeft": 5,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "flexDirection": "row",
        "flexWrap": "wrap",
      }
    }
  >
    <SelectedUser
      onRemove={[MockFunction]}
      teammateNameDisplay="full_name"
      theme={
        Object {
          "awayIndicator": "#ffbc42",
          "buttonBg": "#166de0",
          "buttonColor": "#ffffff",
          "centerChannelBg": "#ffffff",
          "centerChannelColor": "#3d3c40",
          "codeTheme": "github",
          "dndIndicator": "#f74343",
          "errorTextColor": "#fd5960",
          "linkColor": "#2389d7",
          "mentionBj": "#ffffff",
          "mentionColor": "#145dbf",
          "mentionHighlightBg": "#ffe577",
          "mentionHighlightLink": "#166de0",
          "newMessageSeparator": "#ff8800",
          "onlineIndicator": "#06d6a0",
          "sidebarBg": "#145dbf",
          "sidebarHeaderBg": "#1153ab",
          "sidebarHeaderTextColor": "#ffffff",
          "sidebarText": "#ffffff",
          "sidebarTextActiveBorder": "#579eff",
          "sidebarTextActiveColor": "#ffffff",
          "sidebarTextHoverBg": "#4578bf",
          "sidebarUnreadText": "#ffffff",
          "type": "Mattermost",
        }
      }
      user={
        Object {
          "id": "userId1",
        }
      }
    />
  </View>
</View>
`;

exports[`SelectedUsers should match snapshot to show warning for ability to add one more user 1`] = `
<View
  style={
    Object {
      "marginBottom": 5,
      "marginLeft": 5,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "flexDirection": "row",
        "flexWrap": "wrap",
      }
    }
  >
    <SelectedUser
      onRemove={[MockFunction]}
      teammateNameDisplay="full_name"
      theme={
        Object {
          "awayIndicator": "#ffbc42",
          "buttonBg": "#166de0",
          "buttonColor": "#ffffff",
          "centerChannelBg": "#ffffff",
          "centerChannelColor": "#3d3c40",
          "codeTheme": "github",
          "dndIndicator": "#f74343",
          "errorTextColor": "#fd5960",
          "linkColor": "#2389d7",
          "mentionBj": "#ffffff",
          "mentionColor": "#145dbf",
          "mentionHighlightBg": "#ffe577",
          "mentionHighlightLink": "#166de0",
          "newMessageSeparator": "#ff8800",
          "onlineIndicator": "#06d6a0",
          "sidebarBg": "#145dbf",
          "sidebarHeaderBg": "#1153ab",
          "sidebarHeaderTextColor": "#ffffff",
          "sidebarText": "#ffffff",
          "sidebarTextActiveBorder": "#579eff",
          "sidebarTextActiveColor": "#ffffff",
          "sidebarTextHoverBg": "#4578bf",
          "sidebarUnreadText": "#ffffff",
          "type": "Mattermost",
        }
      }
      user={
        Object {
          "id": "userId2",
        }
      }
    />
    <SelectedUser
      onRemove={[MockFunction]}
      teammateNameDisplay="full_name"
      theme={
        Object {
          "awayIndicator": "#ffbc42",
          "buttonBg": "#166de0",
          "buttonColor": "#ffffff",
          "centerChannelBg": "#ffffff",
          "centerChannelColor": "#3d3c40",
          "codeTheme": "github",
          "dndIndicator": "#f74343",
          "errorTextColor": "#fd5960",
          "linkColor": "#2389d7",
          "mentionBj": "#ffffff",
          "mentionColor": "#145dbf",
          "mentionHighlightBg": "#ffe577",
          "mentionHighlightLink": "#166de0",
          "newMessageSeparator": "#ff8800",
          "onlineIndicator": "#06d6a0",
          "sidebarBg": "#145dbf",
          "sidebarHeaderBg": "#1153ab",
          "sidebarHeaderTextColor": "#ffffff",
          "sidebarText": "#ffffff",
          "sidebarTextActiveBorder": "#579eff",
          "sidebarTextActiveColor": "#ffffff",
          "sidebarTextHoverBg": "#4578bf",
          "sidebarUnreadText": "#ffffff",
          "type": "Mattermost",
        }
      }
      user={
        Object {
          "id": "userId1",
        }
      }
    />
  </View>
  <FormattedText
    defaultMessage="You can add 1 more user"
    id="mobile.more_dms.one_more"
    style={
      Object {
        "color": "rgba(61,60,64,0.6)",
        "fontSize": 12,
        "marginBottom": 2,
        "marginRight": 5,
        "marginTop": 10,
      }
    }
  />
</View>
`;
